### PR TITLE
split ErrInvalidKey into ErrInvalidType and ErrInvalidKey 

### PR DESCRIPTION
**Pull request by martinlindhe**
to give more information of what goes wrong to the user

### CODE REVIEW

File: errors.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 // Error constants
 var (
 	ErrInvalidKey       = errors.New("key is invalid or of invalid type")
+	ErrInvalidType      = errors.New("key is not []byte")
```
Comment (dgrijalva) [inline errors.go]: This is not correct. Only `HSXXX` keys accept `[]byte`.

**Comment by martinlindhe** [comment]
Sorry about that. I made the message more like the original intent. Hope this works better.

**Comment by martinlindhe** [comment]
is this good to go, or anything more needs to be done?

**Comment by dgrijalva** [comment]
Lgtm. This will have to be part of 3.0 as it contains breaking API changes. 

-dave

**Comment by dgrijalva** [comment]
Could we make the constant `ErrInvalidKeyType`?

**Comment by dgrijalva** [comment]
Landed in release_3_0_0

**Pull request closed without merging**